integration-tests/src/test/scala/za/co/absa/spline/harvester/LineageHarvesterSpec.scala

```diff
@@ -218,6 +226,56 @@ class LineageHarvesterSpec extends FlatSpec with Matchers with SparkFixture with
         assertDataLineage(expectedOperations, expectedAttributes, plan)
       }
     })
+
+  "Create table as" should "produce lineage when creating a Hive table from temp view" taggedAs ignoreIf(ver"$SPARK_VERSION" < ver"2.3") in
+    withRestartingSparkContext {
+      withCustomSparkSession(_
+        .enableHiveSupport()
+        .config("hive.exec.dynamic.partition.mode", "nonstrict")) { spark =>
+
+        val databaseName = s"unitTestDatabase_${this.getClass.getSimpleName}"
+        withHiveDatabase(spark)(databaseName) {
+
+          val (plan, _)= withLineageTracking(spark) { lineageCaptor => {
+
+            import spark.implicits._
+
+            val df = spark.createDataset(Seq(TestRow(1, 2.3, "text")))
+
+            lineageCaptor.lineageOf {
+              df.createOrReplaceTempView("tempView")
+              spark.sql("create table users_sales as select * from tempView ")
+            }
+          }}
+
+          val writeOperation = plan.operations.write
+          writeOperation.id shouldEqual 0
+          writeOperation.append shouldEqual false
+          writeOperation.childIds shouldEqual Seq(1)
+          writeOperation.params("destinationType") shouldEqual Some("hive")
+
+          val otherOperations = plan.operations.other.sortBy(_.id)
+
+          val firstOperation = otherOperations(0)
+          firstOperation.id shouldEqual 1
+          firstOperation.childIds shouldEqual Seq(2)
+          firstOperation.params("name") shouldEqual "Project"
+
+          val secondOperation = otherOperations(1)
+          secondOperation.id shouldEqual 2
+          secondOperation.childIds shouldEqual Seq(3)
+          if (ver"$SPARK_VERSION" < ver"2.4")
+            secondOperation.params("name") shouldEqual "SubqueryAlias"
+          else
+            secondOperation.params("name") shouldEqual Some("`tempview`")
```

Comment:
Common unit tests design practices suggest to avoid `if-else` statements as well as other types of branching in general. The reason is that it puts cognitive load onto the reader, while unit tests are supposed to be easily readable.
I'd suggest using disjunctive assertion instead of branching.
```suggestion
            secondOperation.params("name") should (equal("SubqueryAlias") or equal("tempview"))
```